fs/xfstests.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+# Based on code by Cleber Rosa <crosa@redhat.com>
+#   copyright: 2011 Redhat
+#   https://github.com/autotest/autotest-client-tests/tree/master/xfstests
+
+
+import os
+import glob
+import re
+import shutil
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process, build, git
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Xfstests(Test):
+
+    def setUp(self):
+        """
+        Build xfstest
+        Source: git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git
+        """
+        sm = SoftwareManager()
+        packages = ['xfslibs-dev', 'uuid-dev', 'libtool-bin', 'e2fsprogs',
+                    'automake', 'gcc', 'libuuid1', 'quota', 'attr',
+                    'libattr1-dev', 'make', 'libacl1-dev', 'xfsprogs',
+                    'libgdbm-dev', 'gawk', 'fio', 'dbench', 'uuid-runtime']
```

Comment:
No Fedora support? Avocado claims to support Fedora, so it would be nice if you could make the tests to work on Fedora as well. I see a packages list fro Fedora in `xfstests` README file, so you could use that as reference.